### PR TITLE
refactor: make root workspace bzlmod only

### PR DESCRIPTION
**Pull request by gregmagolan**
Reduce the maintenance burden caused by duplication of Bazel deps in the root workspace.

CI coverage for WORKSPACE is still in-place in all the e2e workspaces.

---

### Test plan

- Covered by existing test cases

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -1,7 +1,7 @@
 """This module contains the project repository dependencies.
 """
 
-load("@bazel_gazelle//:deps.bzl", "go_repository")
```
Comment (jbedard) [inline deps.bzl:4]: This means previously we were fetching it twice?

File: WORKSPACE.bazel [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# An external repository for test to use
```
Comment (jbedard) [inline WORKSPACE.bazel:1]: Why is this file still needed at all?

File: deps.bzl (hunk 1 above)
Comment (gregmagolan) [reply deps.bzl:4]: Yes, we likely were under bzlmod

File: WORKSPACE.bazel (hunk 2 above)
Comment (kormide) [reply WORKSPACE.bazel:1]: Should this be `WORKSPACE.bzlmod`?

Comment (gregmagolan) [reply WORKSPACE.bazel:1]: I chatted with Alex and that and the answer is that you should _only_ use `WORKSPACE.bzlmod` in workspaces that you want to support both bzlmod and non-bzlmod. The go-to is WORKSPACE.bazel if you only support one or the other.

**Review by jbedard — APPROVED** [review]

File: WORKSPACE.bazel (hunk 2 above)
Comment (jbedard) [reply WORKSPACE.bazel:1]: I think we should still be able to drop the go deps from here and use pure bzlmod for go. We'd still have to generate `deps.bzl` for non-bzlmod repos (such as e2e tests) to use, but bazel-lib istelf would never use it.

That can be followup later though 🤷
